### PR TITLE
retroarch: update to 1.7.9.2

### PR DESCRIPTION
**Pull request by cmitu**
_This is a small modification on top of #2853, which was reverted because of a bug in RetroArch's build system._
Updates the RetroArch to version 1.7.9 (from the _v1.7.9.2_ release tag). 
Some changes since 1.7.6:
 - there's a new notification UI (menu widgets) which is the default. 
 - the joypad sorting is dropped upstream, so the patch included in RetroPie is not needed anymore. Remains to be seen if the new joypad order is compatible with the previous behavior.
 - upstream dropped the `video_shader` parameter and replaced it instead with a CLI option (`--set-shader`), but in my tests this is not behaving the same - the shader from the command line overrides any game overrides, whereas before, the shader saved for the game would override the one in the config file. Also, for some reason the default for enabling the Video Shaders has been changed to _false_.    
I added a patch to bring back reading the `video_shader` parameter when every other auto-shader preset (game/core/content/global) is not present.  
- there's a new `quit_press_twice` option, defaults to `true`, which expects the user to press the Quit key/hotkey combo twice to exit.

- the default compile options enable support for OpenGL1, disable them on the Pi when the BRCM libs are used since it produces a linking error (missing `-lGL`).
- disabled `vulkan` and `wayland` for non-X11 platforms (fixes previous PR's build with RetroArch < 1.8.0 and SDL not having Wayland enabled - see https://github.com/libretro/RetroArch/issues/9583).

I rebased the RetroPie patches on-top of the RetroArch's version pulled and added a few new options to be set as in the default config.
- `quit_press_twice`, `video_shader_enable` - these will be also set for migrated configs, since they're unset by default.
-  I added `rgui_browser_directory` to make browsing for files start in RetroPie's rom folder. Useful when multi-disk games are loaded and a new disk must be appended.
- I changed the default font size due to the new notification system - 12 is unreadable with the new *menu widgets* style notifications. The default is 32.
- I've set the new `rgui_aspect_ratio_lock` configuration option to _Integer scale_, this makes the RGUI more readable instead of being stretched to the content's aspect ratio (see https://github.com/libretro/RetroArch/pull/8211).

Things needed:
 - `retroarch-assets.zip` needs to be updated on the RP files server, to include both the `ozone` 
`menu_widgets` folders (https://github.com/libretro/retroarch-assets) and to update the `xmb/monochrome` assets.

### CODE REVIEW

File: scriptmodules/emulators/retroarch.sh [hunk 1]
```diff
@@ -56,14 +55,15 @@ function build_retroarch() {
         params+=(--disable-ffmpeg)
     fi
     isPlatform "gles" && params+=(--enable-opengles)
-    # Temporarily block dispmanx support for fkms until upstream support is fixed
-    isPlatform "dispmanx" && ! isPlatform "kms" && params+=(--enable-dispmanx)
     isPlatform "rpi" && isPlatform "mesa" && params+=(--disable-videocore)
+     # Temporarily block dispmanx support for fkms until upstream support is fixed
```
Comment (joolswills) [inline scriptmodules/emulators/retroarch.sh]: Indent.

**Comment by joolswills** [comment]
Builds fine now btw thanks.

**Comment by joolswills** [comment]
The xmb monochrome folder has the extra assets for the ticks on the on-screen messages. Please can you detail what the other assets are needed for by default.

**Comment by cmitu** [comment]
For the extra assets, we'll need (from https://github.com/libretro/retroarch-assets):
* the `menu_widgets` folder
* the `ozone` folder
* (optional) the `rgui` folder, which has some theme presets and backgrounds that can be used.

**Comment by joolswills** [comment]
Thanks. I meant more specifically what are ozone and menu_widgets needed for (so I can test).

**Comment by cmitu** [comment]
The new assets are for for the new notification interface. Basically any notification message should show up with the new font and icons
- controller config loaded
- shader loaded
- remap applied/saved.

**Comment by cmitu** [comment]
@joolswills should be rebased now. I've added the `rgui_aspect_ratio_lock` also to the default rp-dist config, the other settings options that have been added were already present.

File: scriptmodules/emulators/retroarch.sh [hunk 2]
```diff
@@ -108,7 +108,7 @@ function install_xmb_monochrome_assets_retroarch() {
     local dir="$configdir/all/retroarch/assets"
     [[ -d "$dir/.git" ]] && return
     [[ ! -d "$dir" ]] && mkUserDir "$dir"
-    downloadAndExtract "$__archive_url/retroarch-xmb-monochrome.tar.gz" "$dir"
+    downloadAndExtract "$__binary_base_url/retroarch-xmb-monochrome.tar.gz" "$dir"
```
Comment (joolswills) [inline scriptmodules/emulators/retroarch.sh]: this shouldn't be included here if it's been rebased over latest master

**Comment by joolswills** [comment]
You seem to have 2 commits (second commit mentioning being rebased which shouldn't be in a commit msg) and a commit from master included.

**Comment by joolswills** [comment]
as I want to change the assets maybe it's easiest I rebase this and commit to my fork?

**Comment by joolswills** [comment]
https://github.com/RetroPie/RetroPie-Setup/compare/master...joolswills:retroarch_1.7.9.2?expand=1

**Comment by joolswills** [comment]
If you're happy I can do a new PR (place your info in it - I put it in the commit as I think it's useful having all the changes described in the commit, and not just in the PR).

**Comment by cmitu** [comment]
> If you're happy I can do a new PR (place your info in it - I put it in the commit as I think it's useful having all the changes described in the commit, and not just in the PR).

Sounds good.

**Comment by cmitu** [comment]
Closing in favour of #2941.

**Pull request closed without merging**